src/nyx/arrow_output_stream.cpp

```diff
@@ -12,95 +12,83 @@
   error "Missing the <filesystem> header."
 #endif
 
-std::tuple<bool, std::optional<std::string>> ArrowOutputStream::create_arrow_file(const Nyxus::SaveOption& arrow_file_type,
-                                                         const std::string& arrow_file_path,
-                                                         const std::vector<std::string>& header) {
+namespace Nyxus {
+
+#ifdef USE_ARROW
+    std::tuple<bool, std::optional<std::string>> ArrowOutputStream::create_arrow_file(const Nyxus::SaveOption& arrow_file_type,
+                                                            const std::string& output_path,
+                                                            const std::vector<std::string>& header) {
 
+        this->arrow_file_path_ = output_path;
 
+        auto path = fs::path(output_path).remove_filename().u8string();
 
-    auto valid_extension = [&arrow_file_path](){
-        if(	arrow_file_path != "" && 
-            !fs::is_directory(arrow_file_path)){
-                if( auto ext = fs::path(arrow_file_path).extension(); 
-                    ext == ".arrow" || ext == ".feather" || ext == ".arrow"){
-                        return true;
-                } else {
-                    return false;
-                } 
+        if (path != "" && !fs::exists(path)) {
+            try {        
+                fs::create_directory(path);
+            } catch (std::exception& e) {
+                return {false, e.what()};
             }
-        return false;
-    }(); 
+        } 
```

Comment:
We probably do not this anymore, since the `output_path` should now contain a fully qualified path. It will fail if this is not honored. You have already implemented the logic to get a correct FQP in the helper function. 
If `output_path` is empty, we should return `{false, "No file path provided"}` otherwise keep moving.